### ISSUE TITLE
Feature Request: Add support for Multiplexed accounts as an experimental feature

### ISSUE THREAD

**Issue by shanzzam** [stellar/laboratory#525]
### What problem does your feature solve?
Multiplexed Accounts (CAP27) were added to protocol 13. As an efforts to promote adoption JS and GO SDKs are being updated to support these. We should provide a way for developers to test multiplexed accounts in the Lab. Until this is fully rolled out, make sure the UI notes this is experimental.

### What would you like to see?

- [x] Ability to create muxxed accounts addresses from an account and extra info
- [x] Parse info based on a muxxed address
- [x] Upgrade the SDK and turn on the muxxed accounts feature flag

**Comment by shanzzam** [comment]
Figma file [here](https://www.figma.com/file/MTrwESxt3qhAWrQfedAPB3/Multiplexed-Accounts?node-id=0%3A1).
Keeping this behind an "Experimental Tab" for now.

**Comment by leighmcculloch** [comment]
I have some comments about the following two pages in the Figma file.

1. It looks like the inputs to creating a multiplexed account from an account is only the `G` strkey. The user should also need to input a `uint64` value that we refer to as the ID. The account and ID will be combined to form the multiplexed account `M` strkey.

2. The page 3 below also shows an `XDR Object`. What's the use case for that? I doubt there would be value in providing that. The M strkey is the value developers will use. We don't offer XDR objects base64 encoded for other components, even though provide decoding of them on the View XDR page.

3. The introduction needs tweaking because an M address doesn't resolve a single G address to many different IDs, an M address resolves to a G address and an ID. The term resolves here has some other meaning that doesn't read to me like what I think we're intending. I suggest rewording this to:

   >A muxed account is an account that multiplexes a single Stellar account by encoding an identifier in the address alongside the Stellar account. A muxed account is an M address that encodes a G address and an identifier.

4. >Create Multiplexed Account
   
   I think in most places we are surfacing these new addresses in code we are calling them Muxed Accounts. Could we use only that terminology to create a single ubiquitous term for these accounts? 

3. >Create subaccount from existing G account address
   
   I think we should reword this to match the terminology used on the Create Account page and refer to the `G account address` as the `Public Key`, so that the language we use is consistent and ubiquitous.
   
   I also think the term subaccount implies that there will be an account on the network. The term used in CAP-27, "virtual account," implies that the account does not exist on network.
   
   Can we get away without this text? The heading immediately above `Created Multiplexed Account` probably says all we need here?

Figma File Page 2:
<img width="1163" alt="Screen Shot 2021-06-02 at 8 36 46 PM" src="https://user-images.githubusercontent.com/351529/120583036-5326a980-c3e2-11eb-952c-29f6f25a8c61.png">

Figma File Page 3:
<img width="1163" alt="Screen Shot 2021-06-02 at 8 36 57 PM" src="https://user-images.githubusercontent.com/351529/120583042-55890380-c3e2-11eb-93b4-ebcf37821641.png">

---

> Keeping this behind an "Experimental Tab" for now.

What's the motivation for putting this under an experimental tab?

**Comment by shanzzam** [comment]
@leighmcculloch Thanks for reviewing this! I'll address each here.

1) I was initially looking at createSubAccount in the Account class but I understand we want to use the [MuxedAccount](https://stellar.github.io/js-stellar-sdk/MuxedAccount.html) instead. I'll create a field for the string input. 

2) `| The page 3 below also shows an XDR Object. What's the use case for that?` 
If this isn't useful we can strike it. I was providing output for properties of the MuxedAccount returned in the [parse](https://stellar.github.io/js-stellar-sdk/MuxedAccount.html#.fromAddress). Although I did leave out the base account and sequence number properties as I thought they were redundant. 

3) `| The introduction needs tweaking`
I'll change the wording to what you provided. The current text was lifted right off the SDK and was placeholder to expand on.

4) RE: Multiplexed vs Muxed Account terminology: The [SEP](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0023.md) and [CAP](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0027.md) refer to MuxedAccount when referencing code but not "Muxed Account", otherwise they are Multiplexed Accounts. I can see where in this instance we can use MuxedAccount, but in the intro I would refer to them as Multiplexed Accounts. 

5) I think what would work better here would be to remove the heading "Create Multiplexed Account" with something else. The two actions on the page are to create an account and parse the account. I'll change "Create subaccount from existing G account address" to Create MuxedAccount from Public key". I'll also change the wording "Get MuxedAccount from M address" to "Get MuxedAccount from M Address"

6) `| What's the motivation for putting this under an experimental tab?`
We aren't advocating for mass adoption of Multiplexed Accounts until we do education/outreach and see wallet support. Given that, it's a good idea to add this disclaimer in the intro.

**Comment by leighmcculloch** [comment]
> 1. createSubAccount in the Account class

Ah, interesting. We probably shouldn't have used the 'subaccount' terminology there either. MuxedAccounts may be virtual accounts, but that is just one use case, they can also be invoices and other things. I opened an issue to fix that: https://github.com/stellar/js-stellar-base/issues/433.

> 4\. The [SEP](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0023.md) and [CAP](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0027.md) refer to MuxedAccount when referencing code but not "Muxed Account", otherwise they are Multiplexed Accounts.

What value do we get out of using Multiplexed as the noun for this in some places and Muxed in others? I think it's fine if we use the term multiplexed to describe what a Muxed Account is, but I think we need to weigh the value in using multiple different names/nouns for a concept. Building a ubiquitous language is really helpful for developers and reduces the overall complexity of building on Stellar. Every time we add a new feature to Stellar it gets more complex. Every time we use multiple terms for that new thing we multiply the complexity and the ambiguity. It's not just code, but the Horizon API refers to these accounts as muxed, and it will be likely that explorers will follow suit since explorers typically use the terms our SDKs use.

> 6\. `| What's the motivation for putting this under an experimental tab?`
>     We aren't advocating for mass adoption of Multiplexed Accounts until we do education/outreach and see wallet support. Given that, it's a good idea to add this disclaimer in the intro.

Got it. Did we consider placing these tools where we'd expect them to go long term, with a label/description that indicates they are experimental? A disadvantage of placing tools on an experimental tab is that they are likely to move when they are no longer experimental. This will create more work when it comes time to make them not experimental, and will require folks using the tool to rediscover the tool once it moves.

**Comment by shanzzam** [comment]
> 4. The SEP and CAP refer to MuxedAccount when referencing code but not "Muxed Account", otherwise they are Multiplexed Accounts.
Sorry I wasn't clear. I'm advocating to only use either "Multiplexed Account" or "MuxedAccount" in Lab or any future tool we add support to. This would follow convention already established in code and SEP/CAP. 
Are you advocating to use "Muxed Account" in any reference materials or tools we create?

RE: Putting this behind the Experimental tab:
Those are good points you make about visibility and having to rediscover them again. The 2 changes would be: Create Account tab would now show the ability to create a MuxedAccount, and "Getting MuxedAccount from M address" could be achieved using the Account endpoint. I'll have another look at the UI and get some more feedback. I'm not too worried about the work to wrap these back into regular UI although @quietbits may have a different opinion :)

**Comment by leighmcculloch** [comment]
> Are you advocating to use "Muxed Account" in any reference materials or tools we create?

Yup, because SDKs and APIs already use this term. It would be very difficult at this point to change those things to use the term multiplexed.

**Comment by Shaptic** [comment]
We should, at the very least, inform people in the documentation that muxed is slang for multiplexed; both terms are rather technical and definitely not in the common vernacular.

**Comment by quietbits** [comment]
I like the idea of putting these features where they will end up in the future. It wouldn't be that much work for us to move them but it will be much easier for users to find them (not wonder where they went later on).

**Comment by shanzzam** [comment]
updated figma file
![updated_lab_create_accounts](https://user-images.githubusercontent.com/5062622/121437703-9380ad00-c937-11eb-89a0-c11ada9f896c.png)

**Comment by shanzzam** [comment]
@quietbits as turning on the muxxed accounts feature flag will turn on support in operations and transactions, please add validation in form input fields where M accounts could be provided. This should be anywhere a G address is valid.

**Comment by shanzzam** [comment]
@quietbits please also update the "Experimental" syntax in the Create Muxed Account tab.

**Comment by quietbits** [comment]
@shanzzam Will do! Sorry I missed the updated Figma design. 😬

**Comment by shanzzam** [comment]
Will review 6/22

**Comment by shanzzam** [comment]
Review passed. @quietbits  Ready for merge.
